apps/src/lib/util/analyticsUtils.js

```diff
@@ -0,0 +1,24 @@
+/* global appOptions */
+import {queryParams} from '@cdo/apps/code-studio/utils';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+
+export const reportTeacherReviewingStudentDslLevel = (
+  additionalPayload = {}
+) => {
+  if (!appOptions) {
+    return;
+  }
+  if (
+    appOptions.readonlyWorkspace &&
+    !appOptions.submitted &&
```

Comment:
I'm interested in this not submitted part- can you explain why we aren't logging for submitted work?

Reply:
The short answer is I copied that logic from [here](https://github.com/code-dot-org/code-dot-org/blob/7b24ae3c7fd5693de4e9b6918cb5171870438010/apps/src/code-studio/levels/multi.js#L146). I'm trying to find a more convincing answer though....

Reply:
Okay, I think it comes from [this conditional](https://github.com/code-dot-org/code-dot-org/blob/1ccbf87fd0297d78b9a83c4e20f20c27ae27c5d0/dashboard/app/controllers/script_levels_controller.rb#L331) where, if someone is viewing another's work, `submitted` isn't set, but if they're viewing their own and can change it `submitted` is true

Reply:
I should probably check that the user isn't in code review mode though